stubs/pyinstaller/PyInstaller/lib/modulegraph/modulegraph.pyi

```diff
@@ -0,0 +1,39 @@
+from typing import Any, Protocol
+from typing_extensions import LiteralString
+
+class _SupportsGraphident(Protocol):
+    graphident: str
+
+# TODO: For typing purposes, once #5768 is complete, it'll be easier to use the modulegraph package directly.
+
+# code, filename and packagepath are always initialized to None. But they can be given a value later.
+class Node:
+    # Compiled code. See stdlib.builtins.compile
+    code: Any | None
+    filename: LiteralString | None
+    graphident: LiteralString
+    identifier: LiteralString
+    packagepath: LiteralString | None
+    def __init__(self, identifier: LiteralString) -> None: ...
+    def is_global_attr(self, attr_name: str) -> bool: ...
+    def is_submodule(self, submodule_basename: str) -> bool: ...
+    def add_global_attr(self, attr_name: str) -> None: ...
+    def add_global_attrs_from_module(self, target_module: Node) -> None: ...
+    def add_submodule(self, submodule_basename: str, submodule_node: Node) -> None: ...
+    def get_submodule(self, submodule_basename: str) -> Node: ...
+    def get_submodule_or_none(self, submodule_basename: str) -> Node | None: ...
+    def remove_global_attr_if_found(self, attr_name: str) -> None: ...
+    def __lt__(self, other: _SupportsGraphident) -> bool: ...
```

Comment:
It's useful to include `__eq__` and `__ne__` overrides, as mypy special-cases them for things like its `--strict-equality` check (included by default if you use `mypy --strict`)

```suggestion
    def __eq__(self, other: object) -> bool: ...
    def __ne__(self, other: object) -> bool: ...
    def __lt__(self, other: _SupportsGraphident) -> bool: ...
```